behaviors/services/signing-service.md

```diff
@@ -0,0 +1,47 @@
+# Signing service
+
+Signing service is a service that contains node private key and allows to sign certain payload with this key.
+
+It preferably runs as a separate process in a separate container. The idea behind it is secure isolation of private key from other parts of the system.
+
+## Interacts with services
+
+* `Conseusus Algo` - uses it to sign blocks.
+* `Transaction Pool` - uses it to sign transaction forwarding.
+
+## APIs
+
+V1:
+
+* `Sign` - signs sha256 hash.
+
+V2:
+
+* `SignTransactionBatch` - signs transactions batch hash.
```

Comment:
why does it need to know what it signs?

Reply:
Agree, modified to signing a data, requires a signature scheme change in order to ensure that a virtual chain may only sign data related to the chain (consensus messages, relay messages, etc)